scripts/dts/edtlib.py

```diff
@@ -1545,25 +1545,56 @@ def _merge_includes(self, raw, binding_path):
             return raw
 
         include = raw.pop("include")
-        fnames = []
+
+        # a list of (name, allowlist, blocklist) tuples
+        include_info = []
+
+        # Simple scalar string case
         if isinstance(include, str):
-            fnames.append(include)
+            include_info.append((include, None, None))
         elif isinstance(include, list):
-            if not all(isinstance(elem, str) for elem in include):
+        # List of strings
+            if all(isinstance(elem, str) for elem in include):
+                for elem in include:
+                    include_info.append((elem, None, None))
+        # List of 'name' maps
+            elif all(isinstance(elem, dict) for elem in include) and \
+                 all(isinstance(elem.get("name"), str) for elem in include):
+                for elem in include:
+                    name = elem["name"]
+                    allowlist = elem.get("property-allowlist")
+                    blocklist = elem.get("property-blocklist")
+                    if blocklist and allowlist:
+                        _err(f"include: in {binding_path} should not specify "
+                             "both 'property-allowlist:' and 'property-blocklist:'")
+                    if allowlist and not isinstance(allowlist, list):
+                        _err(f"'property-allowlist' in {binding_path} should "
+                             f"be a list: {allowlist}")
+                    if blocklist and not isinstance(blocklist, list):
+                        _err(f"'property-blocklist' in {binding_path} should "
+                             f"be a list: {blocklist}")
+                    include_info.append((name, allowlist, blocklist))
+            else:
                 _err(f"all elements in 'include:' in {binding_path} "
-                     "should be strings")
-            fnames += include
+                     "should either be strings or maps with 'name' key')")
         else:
             _err(f"'include:' in {binding_path} "
-                 "should be a string or a list of strings")
+                 "should be a string, list of strings, or list of dicts "
+                 "with 'name' keys")
 
         # First, merge the included files together. If more than one included
         # file has a 'required:' for a particular property, OR the values
         # together, so that 'required: true' wins.
 
         merged = {}
-        for fname in fnames:
-            _merge_props(merged, self._load_raw(fname), None, binding_path,
+        for (fname, allowlist, blocklist) in include_info:
+            raw_included = self._load_raw(fname)
+            for k in list(raw_included["properties"].keys()):
```

Comment:
'properties' isn't always in raw_included. See e.g. dts/bindings/ethernet/ethernet,fixed-link.yaml, which only has a child-binding. Need to handle this.

Reply:
Fixed.